plugins/mac-capture/window-utils.m

```diff
@@ -23,14 +23,25 @@
 	return [o1[WINDOW_NUMBER] compare:o2[WINDOW_NUMBER]];
 };
 
+NSArray *filter_nonzero_kcgwindowlayer_sources(NSArray *windows_arr)
+{
+	NSPredicate *pred =
+		[NSPredicate predicateWithFormat:@"(kCGWindowLayer == 0)"];
+	NSArray *new_windows_arr =
+		[windows_arr filteredArrayUsingPredicate:pred];
+
+	return new_windows_arr;
+}
+
 NSArray *enumerate_windows(void)
 {
 	NSArray *arr = (NSArray *)CGWindowListCopyWindowInfo(
 		kCGWindowListOptionOnScreenOnly, kCGNullWindowID);
+	NSArray *filtered_arr = filter_nonzero_kcgwindowlayer_sources(arr);
```

Comment:
`arr` is autoreleased below.  Should `filtered_arr` also be autoreleased?

Reply:
@RytoEX In Objective-C only objects that we own need to be released. According to [official documentation](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html), this consists of objects we create using `new`, `init`, `copy`, or `mutableCopy`. We do not perform any of these actions to create `filtered_arr`, so we do not need to release it. In fact, calling either `[filtered_arr release]` or `[filtered_arr autorelease]` will attempt to free memory that's not there, resulting in OBS crashing because of a double free scenario.

I'm not too familiar with the internals of `CGWindowListCopyWindowInfo` but I'm guessing there's a `new` or `init` involved in its returned `NSArray`, which may be why it needs to be marked with `autorelease`.

Reply:
Perfectly valid to explain that much to someone who doesn't develop on macOS.  I understand the situation now.